source/extensions/omni.isaac.orbit/omni/isaac/orbit/devices/gamepad/se2_gamepad.py

```diff
@@ -0,0 +1,155 @@
+        self.v_x_sensitivity = v_x_sensitivity
+        self.v_y_sensitivity = v_y_sensitivity
+        self.omega_z_sensitivity = omega_z_sensitivity
+        # acquire omniverse interfaces
+        self._appwindow = omni.appwindow.get_default_app_window()
+        self._input = carb.input.acquire_input_interface()
+        self._gamepad = self._appwindow.get_gamepad(0)
+        self._gamepad_sub = self._input.subscribe_to_gamepad_events(self._gamepad, self._on_gamepad_event)
+        # bindings for gamepad to command
+        self._create_key_bindings()
+        # command buffers
+        self._base_command = np.zeros(3)
+        # dictionary for additional callbacks
+        self._additional_callbacks = dict()
+
+    def __str__(self) -> str:
+        """Returns: A string containing the information of joystick."""
+        msg = f"Gamepad Controller for SE(2): {self.__class__.__name__}\n"
+        msg += "\tMove in X-Y plane: left stick\n"
+        msg += "\tRotate in Z-axis: right stick\n"
+        return msg
+
+    """
+    Operations
+    """
+
+    def reset(self):
+        # default flags
+        self._base_command.fill(0.0)
+
+    def add_callback(self, key: str, func: Callable):
+        """Add additional functions to bind gamepad.
+
+        A list of available keys are present in the
+        `carb documentation <https://docs.omniverse.nvidia.com/kit/docs/carbonite/latest/docs/python/carb.html?highlight=keyboardeventtype#carb.input.GamepadInput>`.
+
+        The callback function should not take any arguments.
+
+        Args:
+            key (str): The gamepad button to check against.
+            func (Callable): The function to call when key is pressed.
+        """
+        self._additional_callbacks[key] = func
+
+    def advance(self) -> np.ndarray:
+        """Provides the result from keyboard event state.
+
+        Returns:
+            np.ndarray -- A 3D array containing the linear (x,y) and angular velocity (z).
+        """
+        return self._base_command
+
+    """
+    Internal helpers.
+    """
+
+    def _on_gamepad_event(self, event: carb.input.GamepadEvent, *args, **kwargs):
+        """Subscriber callback to when kit is updated.
+
+        Reference:
+            https://docs.omniverse.nvidia.com/kit/docs/carbonite/latest/docs/python/carb.html?highlight=keyboardeventtype#carb.input.GamepadInput
+        """
+        self.reset()  # the base command depend only on the current gamepad status
+
+        cur_val = event.value
+        absval = abs(event.value)
+
+        # Ignore 0 since it signifies the movement  of the stick has stopped,
+        # but doesn't mean it's at center...could be being held steady
+        if absval == 0:
+            return True
+        if absval < self.deadzone:
+            cur_val = 0
+
+        if event.input in self._INPUT_KEY_VALUE_MAPPING:
+            self._base_command += self._INPUT_KEY_VALUE_MAPPING[event.input] * cur_val
+        # remove the command when un-pressed
```

Comment:
As a practice, it is better to not have this docstring between the if-else. Either move it to the top explaining what you're checking, or move it inside the else statement.

Reply:
Thanks for your comment. I removed this docstring in the new commit. It is because that there is no press or un-press events when it comes to the gamepad.